### PR TITLE
Move determineErrorMessage func to function package

### PR DESCRIPTION
**Pull request by elsteelbrain**

### CODE REVIEW

File: function/function.go [hunk 1]
```diff
@@ -222,3 +222,26 @@ func (f *Function) callEmulator(payload []byte) ([]byte, error) {
 	defer resp.Body.Close()
 	return ioutil.ReadAll(resp.Body)
 }
+
+func DetermineErrorMessage(err error) string {
```
Comment (mthenw) [inline function/function.go]: doesn't have to be public function

File: router/router.go [hunk 2]
```diff
@@ -239,7 +238,7 @@ func (router *Router) handleHTTPEvent(event *eventpkg.Event, w http.ResponseWrit
 		event.Data = httpdata
 		resp, err := router.callFunction(space, *backingFunction, *event)
 		if err != nil {
-			message := determineErrorMessage(err)
+			message := function.DetermineErrorMessage(err)
```
Comment (mthenw) [inline router/router.go]: My comment also meant that you should only use error message returned by callFunction. Not only moving the function from another package.

File: function/function.go (hunk 1 above)
Comment (elsteelbrain) [reply function/function.go]: How do you mean? It is being used in a different package (`router`). So, doesnt it have to be?

**Comment by elsteelbrain** [comment]
@mthenw Right now, it reports a not found function with error message `unable to look up registered function`. Do we want to show this error or do we want the generic `Function call failed. Please check logs.`?

**Comment by mthenw** [comment]
@elsteelbrain I don't fully understand why the tests are failing. They should be passing as it was before because actual logic didn't change, right?

**Comment by elsteelbrain** [comment]
They're failing because previously we were determining messages for all error, no matter where they came from. We had cases, if the error didn't fit our case, it would send general message `Function call failed. Please check logs. `. Now, I've changed the  `ErrFunctionAccessDenied` error messages from `function` package at their source.

File: function/function.go [hunk 3]
```diff
@@ -142,10 +142,20 @@ func (f *Function) callAWSLambda(payload []byte) ([]byte, error) {
...
+						"lambda:invokeFunction permission to these credentials")}
+			case "UnrecognizedClientException":
+				return nil, &ErrFunctionAccessDenied{errors.New(
+					"function call failed with UnrecognizedClientException. The provided credentials " +
+						"are invalid. Please provide valid credentials")}
+			case "ExpiredTokenException":
+				return nil, &ErrFunctionAccessDenied{errors.New(
+					"function call failed with ExpiredTokenException. The provided security token for " +
+						"the function has expired. Please provide an updated security token or provide " +
+						"permanent credentials")}
```
Comment (alexdebrie) [inline function/function.go]: @elsteelbrain Nit, but can you capitalize the first word in each of these three errors?

Comment (mthenw) [reply function/function.go]: I will take care of that. I have some errors cleanup in separate branch. We can close it.

**Pull request closed without merging**